lib/matplotlib/spines.py

```diff
@@ -500,39 +424,61 @@ def get_position(self):
     def get_spine_transform(self):
         """Return the spine transform."""
         self._ensure_position_is_set()
-        what, how = self._spine_transform
-
-        if what == 'data':
-            # special case data based spine locations
-            data_xform = self.axes.transScale + \
-                (how + self.axes.transLimits + self.axes.transAxes)
-            if self.spine_type in ['left', 'right']:
-                result = mtransforms.blended_transform_factory(
-                    data_xform, self.axes.transData)
-            elif self.spine_type in ['top', 'bottom']:
-                result = mtransforms.blended_transform_factory(
-                    self.axes.transData, data_xform)
-            else:
-                raise ValueError('unknown spine spine_type: %s' %
-                                 self.spine_type)
-            return result
 
+        position = self._position
+        if isinstance(position, str):
+            if position == 'center':
+                position = ('axes', 0.5)
+            elif position == 'zero':
+                position = ('data', 0)
+        assert len(position) == 2, 'position should be 2-tuple'
+        position_type, amount = position
```

Comment:
Trying to read through this, and `amount` could do with a more descriptive name I think.

Reply:
The variable name comes from the old version, any suggestions?

Reply:
@efiring proposed "where" and "location"; you get to pick one or propose another :)